src/python/pants/init/logging.py

```diff
@@ -140,15 +95,41 @@ def trace(self, message, *args, **kwargs):
         LogLevel.TRACE.set_level_for(requests_logger)
         requests_logger.propagate = True
 
-    if not log_dir:
-        return None
+
+def setup_logging_to_stderr(
+    level: LogLevel, *, warnings_filter_regexes: Optional[List[str]] = None
+) -> None:
+    """Sets up Python logging to stderr, proxied to Rust via a NativeHandler.
+
+    We deliberately set the most verbose logging possible (i.e. the TRACE log level), here, and let
+    the Rust logging faculties take care of filtering.
+    """
+    _common_logging_setup(level, warnings_filter_regexes)
+
+    python_logger = logging.getLogger(None)
+    handler = NativeHandler(level, stream=sys.stderr)
+    python_logger.addHandler(handler)
+    LogLevel.TRACE.set_level_for(python_logger)
+
+
+def setup_logging_to_file(
+    level: LogLevel,
+    *,
+    log_dir: str,
+    log_filename: str = "pants.log",
+    warnings_filter_regexes: Optional[List[str]] = None,
+) -> NativeHandler:
```

Comment:
Consider a docstring. For example, why does this not set the level to TRACE?

Reply:
I'm trying to not change any previous logging behavior in this refactor commit, but this is a good question actually. I'm not sure why the code that logs to a file doesn't set the Python level to TRACE when the code that logs to stderr does. @jsirois ? @stuhood ?